UPGRADE.md

```diff
@@ -0,0 +1,50 @@
+# Upgrade
+
+This document here is intended to simplify upgrading to newer versions by extending the changelog.
+
+## 0.21 -> 0.22
+
+0.22 changes how to define NIFs. Users upgrading to 0.22 should to do two things:
+
+1. Replace `rustler::rustler_export_nifs!` with `rustler::init!`
```

Comment:
Maybe also note the change in `atoms!` and maybe resource/struct/etc?

Reply:
I think `atoms!` and `resource` are quite well described in the changelog, but I added more information on `rustler::init!`, as well as how to migrate schedule flags and renames. What do you think?

(Also, thanks for reading this PR! :))

Reply:
I think it's good! I think keeping it all in one place will save some time for those looking to upgrade :) Great work 👍